openvalidators/reward/dpo.py

```diff
@@ -33,17 +33,23 @@ def name(self) -> str: return RewardModelType.dpo.value
     def __init__(self, device: str):
         super().__init__()
         self.device = device
+        self.penalty = 1.2
         self.tokenizer = AutoTokenizer.from_pretrained(DirectPreferenceRewardModel.reward_model_name)
         self.model = AutoModelForCausalLM.from_pretrained(DirectPreferenceRewardModel.reward_model_name,
                                                           trust_remote_code=True,
                                                           torch_dtype=torch.float16).to(self.device)
 
-    def reward_single(self, prompt: str, completion: str, name: str) -> float:
+    def reward_single(self, prompt: str, completion: str, name: str ,with_penalty=True) -> float:
         r""" Calculates a direct preference optimization (DPO) style reward for a completion,
         which is a reference model's average log-probability for completion tokens given a prompt.
         Uses guidance from https://github.com/eric-mitchell/direct-preference-optimization/blob/main/trainers.py.
         """
         with torch.no_grad():
+
+            # Check if completion is 
+            if completion.strip() == '' or len(completion) <= 5:
+                return -11 # exp(-11)=1.67e-5 < 2e-5=1/50257 (typical vocab size)
```

Comment:
I'm not sure if I got it why is -11, could you please elaborate more so I could better understand it?

Reply:
`exp(-11)` corresponds to base value given to zero or short responses; it is the nearest integer value that is less than equal probability value across all logits (1/50257).

Reply:
Would it be feasible to calculate this in runtime by getting something like `1 / model.vocab_size`? That way the code will be independent of the model used as it would be calculated dynamically.

Reply:
> Would it be feasible to calculate this in runtime by getting something like `1 / model.vocab_size`? That way the code will be independent of the model used as it would be calculated dynamically.

Yes this can be done in a future update, and will be necessary if the DPO model tokenizer is changed to something non-standard.

Reply:
Got it, I will create an issue for that so we don't lose track of this